src/fsharp/LexFilter.fs

```diff
@@ -541,6 +564,12 @@ type LexFilterImpl (lightSyntaxStatus: LightSyntaxStatus, compilingFsLib, lexer,
         | Parser.EOF _ -> tokenTup.LexbufState.StartPos.ColumnMinusOne
         | _ -> tokenTup.LexbufState.StartPos 
 
+    //----------------------------------------------------------------------------
+    // TokenTup pool
+    //--------------------------------------------------------------------------
+
+    let pool = TokenTupPool()
```

Comment:
I don't know the answer to this question, but it came it mind:

Is this okay with multiple threads? I'm not 100% clear on how that works with module-bound values, but I know there are some weird things that can happen. Since parsing operations in the IDE happen concurrently instead of serially it came to mind.

Reply:
It's not an issue. This value is inside a class, `LexFilterImpl` and each parsing file gets their own `LexFilterImpl` which means their own `TokenTup` pool. So, if we ever parsed multiple files in parallel, we are safe because each file being parsed has their own `LexFilterImpl` instance.